Add kubernetes-sigs/release-notes jobs

saschagrunert opened the pull request:
This PR adds the release notes jobs including the owner file for [kubernetes-sigs/release-notes](https://github.com/kubernetes-sigs/release-notes). On my local tests with phaino the `$JOB_NAME` environment variable was not exposed, which is hopefully only related to the tooling.

A good-to-test PR would be https://github.com/kubernetes-sigs/release-notes/pull/26

fejta commented on config/jobs/kubernetes-sigs/release-notes/release-notes-presubmits.yaml:
```diff
@@ -0,0 +1,47 @@
+    npm run $JOB_NAME
+
+spec: &spec
+  <<: *config
```
Please follow the style other yaml files in this repo and avoid using advanced yaml functionality like anchors, templating, etc.

These json objects that allow comments.

saschagrunert replied:
Changed it according to the notes.

nikhita commented on config/jobs/kubernetes-sigs/release-notes/release-notes-presubmits.yaml:
```diff
@@ -0,0 +1,93 @@
+              - -c
+              - >
+                npm install &&
+                npm run $JOB_NAME
```
I am absolute noob with npm so sorry for naive question :see_no_evil: but does this require the project to be checked out at `$GOPATH/src/sigs.k8s.io/release-notes` by any chance? By default, the project will be checked out at `$GOPATH/src/github.com/kubernetes/release-notes`.

If yes, this will also need `path_alias: sigs.k8s.io/release-notes`.

I also see that there is some Go code in the repo, but it doesn't use canonical import paths for importing kubernetes projects: https://github.com/kubernetes-sigs/release-notes/blob/e3627c37e54019ae18a889bdce70ff96cdfa7a11/main.go#L28 ...however, the jobs aren't testing Go code here...so I'm not sure how this all fits in, but just wanted to double check. :smile:

saschagrunert replied:
I think there  should be no path adaption needed right now since we do not use the go part of the project yet. :)

nikhita commented:
The bazel test failes because this needs an update in https://github.com/kubernetes/test-infra/blob/master/testgrid/config.yaml.

saschagrunert commented:
> The bazel test failes because this needs an update in https://github.com/kubernetes/test-infra/blob/master/testgrid/config.yaml.

Yeah I'm still trying to figure out what exactly is wrong with the config.yaml content :confused:

nikhita commented on testgrid/config.yaml:
```diff
@@ -1743,6 +1743,29 @@ test_groups:
   num_columns_recent: 20
 
+# kubernetes-sigs/release-notes
+- name: pull-release-notes-build-dev
```
Sorry, I didn't realize you updated it! :woman_facepalming: 

I think this should be name of the job i.e. just `build-dev`.

saschagrunert replied:
:innocent: I must be missing some detail somewhere int this file...

nikhita commented:
Maybe this just needs an update in `testgrid/cmd/configurator/config_test.go`?

nikhita commented on testgrid/config.yaml:
```diff
@@ -1743,6 +1743,29 @@ test_groups:
 
+# kubernetes-sigs/release-notes
+- name: pull-release-notes-build-dev
+  gcs_prefix: kubernetes-sigs/release-notes/pull-release-notes-build-dev
```
This needs to be `kubernetes-jenkins/pr-logs/directory/pull-release-notes-build-dev` (and same for all) :)

saschagrunert replied:
Oh thank you so much for hinting me into the right direction :pray:

fejta commented:
Any chance someone from sig-release can review this?

saschagrunert commented:
> I think all the instances of `npm install` should instead be `npm ci`. We commit `package-lock.json` in the repo and we don't want to regenerate it during any tests.

Changed as suggested and rebased on top of the latest master branch.

jeefy commented:
/lgtm
